src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,769 @@
+                    if (symbolInfo.Symbol == null ||
+                        symbolInfo.Symbol is IMethodSymbol methodSymbol &&
+                        methodSymbol.Parameters.Length == 1)
+                    {
+                        processingState.AddVariableName(queryContinuation.Identifier.ValueText);
+                        processingState.AddStatementProcessingMethod(
+                            s => AddToBlockTop(
+                                 SyntaxFactory.LocalDeclarationStatement(
+                                    SyntaxFactory.VariableDeclaration(
+                                        VarNameIdentifier,
+                                        SyntaxFactory.SingletonSeparatedList(
+                                            SyntaxFactory.VariableDeclarator(
+                                                queryContinuation.Identifier,
+                                                argumentList: null,
+                                                SyntaxFactory.EqualsValueClause(selectClause.Expression))))), s));
+
+                        return TryProcessQueryBody(queryContinuation.Body, processingState);
+                    }
+                }
+
+                return false;
+            }
+
+            private bool TryProcessWhereClause(WhereClauseSyntax whereClause, ProcessingState processingState)
+            {
+                if (IsValidWhereClauseSymbol(whereClause))
+                {
+                    processingState.AddStatementProcessingMethod(
+                         s => SyntaxFactory.Block(
+                             SyntaxFactory.IfStatement(
+                                 ProcessCondition((whereClause).Condition),
+                                 s)));
+
+                    return true;
+                }
+
+                return false;
+            }
+
+            private bool IsValidWhereClauseSymbol(WhereClauseSyntax whereClause)
+            {
+                if (!TryGetMethodSymbol(whereClause, out var methodSymbol))
+                {
+                    return false;
+                }
+
+                var returnType = methodSymbol.ReturnType;
+                if (returnType.Name != "IEnumerable") // TODO better that name check?
+                {
+                    return false;
+                }
+
+                var typeArguments = returnType.GetTypeArguments();
+                if (typeArguments.Length != 1)
+                {
+                    return false;
+                }
+
+                var typeArgument = typeArguments.Single();
+                switch (methodSymbol.Parameters.Length)
+                {
+                    // IEnumerable<T> Where(Func<T, bool> predicate) 
+                    case 1:
+                        return IsPredicate(methodSymbol.Parameters.First().Type, typeArgument);
+
+                    // IEnumerable<T> Where(this IEnumerable<T>, Func<T, bool> predicate) 
```

Comment:
you can call .Constructed from so you always get the orginal static extension form.

Reply:
I tried this but it does not. ConstructedFrom for case 1 is still case 1. 

---
In reply to: [176868754](https://github.com/dotnet/roslyn/pull/25362#discussion_r176868754) [](ancestors = 176868754)